services/QuillLMS/app/controllers/api/v1/activity_sessions_controller.rb

```diff
@@ -104,12 +106,30 @@ def destroy
 
   private def transform_incoming_request
     if params[:concept_results].present?
-      @concept_results = params.delete(:concept_results)
+      @concept_results = params.permit(concept_results_permitted_params).delete(:concept_results).map(&:to_h)
     else
       params.delete(:concept_results)
     end
   end
 
+  private def concept_results_permitted_params
+    {
+      concept_results: [
+        :concept_uid,
+        :question_type,
+        metadata: [
```

Comment:
So I think we need to double check this `metadata` params list for the different activity types. Here's one from Proofreader that includes  `unchanged` and `instructions` keys. Let's look at all of the different question_types.
```
{"answer":"on","prompt":"He put his camera lens \u003cstrong\u003earound/on\u003c/strong\u003e the glass and took one of the most famous photos in history.","correct":1,"unchanged":false,"instructions":"\u003cp\u003eRead the passage. At each underlined part of the passage, choose the correct option and delete the incorrect option and the slash (/).\u003c/p\u003e","questionNumber":6}

Reply:
Hmm, so looking at how many possible params there are, I wonder if it would be better to just allow any keys since that is feature-parity to the old code (note, I'm not entirely sure if this is even possibe, i.e. if you can do `metadata: [{}]`). 

It's less safe than named params (although it is the same as the current code), but I'm slightly worried we might be missing params here. Curious your thoughts (I could be convinced either way).

https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-permit
![Screen Shot 2021-06-21 at 2 27 43 PM](https://user-images.githubusercontent.com/1304933/122810498-6e782c80-d29d-11eb-9c0c-fd436969217c.png)